Framenet: Casting list and dict into sorted for doctests

alvations opened the pull request:
Added a `__lt__` function so that frames (i.e. `AttrDict`) can be sorted by their IDs. 

Making sure that doctests for Framenet are consistent across Python versions. #1901 

Preparing for Py3.6 support #1659

alvations committed:
Casting list and dict into sorted for doctests

Sorted the frames_by_lemmas example

Use ellipsis

alvations committed:
Added the __lt__ function

alvations committed:
Use output sorted output by ID

Sorted outputs for doctest

Sorted output for doctest

Fix typo in doctest

Added PrettyList import

Cast sorted AttrDict into PrettyList for doctest

Cast AttrDict into PrettyList for doctest

alvations commented:
[CI: restest]

alvations commented:
[CI: retest]

stevenbird commented:
Thanks in advance @nschneid :)

nschneid commented on nltk/corpus/reader/framenet.py:
```diff
@@ -686,6 +686,10 @@ def __getitem__(self, name):
 
+    def __lt__(self, other):
+        """ Comparison operator used mainly to sort a list of AttrDict by ID. """
+        return self['ID'] < other['ID']
```
I don't think it's guaranteed that `AttrDict` instances have an ID. But that's only an issue if there is code that attempts to compare or sort them, so this is probably fine.

alvations replied:
Ah now I understand, the `AttrDict` is a generic class. I thought it was a specific object made for Frames.

nschneid commented on nltk/corpus/reader/framenet.py:
```diff
@@ -2242,7 +2247,7 @@ def frame_relation_types(self):
 
         >>> from nltk.corpus import framenet as fn
-        >>> frts = list(fn.frame_relation_types())
+        >>> frts = sorted(list(fn.frame_relation_types()))
```
I think the `list()` is redundant if it's wrapped in `sorted()`.

nschneid reviewed (changes requested):
Looks good, just a minor simplification

stevenbird commented:
Thanks very much @nschneid

alvations committed:
Removed list casting before sorted

alvations committed:
Specific key to sort in doctest

Instead of specifying a key for `__lt__()`, specifying a key to sort on would be better.

alvations committed:
Sort AttrDict with specified key

alvations commented:
Thank you @nschneid for the review!

Specifying a `self["ID"]` in the generic `AttrDict` less than operator would have resulted in KeyError. I've changed the doctest to sort on key using the `operator.itemgetter`. 

I hope that's a better solution for standardizing the output of the doctests for framenet across Python versions.

nschneid commented:
Makes sense to me. Thanks @alvations

stevenbird commented:
Thanks @alvations, @nschneid!
